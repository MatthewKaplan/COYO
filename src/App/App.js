import React, { Component } from 'react';
import Header from '../Header/Header.js';
import Carousel from '../Carousel/Carousel.js';
import TypeCard from '../TypeCard/TypeCard.js';
import search from '../Search/SearchFunction.js';
import SearchBar from '../Search/SearchBar.js';
import Studios from '../Studios/Studios.js';


import '../Styles/Main.scss';

export default class App extends Component {
  constructor() {
    super();
    this.state = {
      searchResults: [],
      studios: [],
      yogaTypes: []
    };
  }

  componentDidMount(){
    fetch('https://fe-apps.herokuapp.com/api/v1/whateverly/1901/SallyHaefling/studios')
      .then(response => response.json())
      .then(yogaStudios => this.setState( {studios: yogaStudios.studios} ))
      .catch(err => console.log(err))
    fetch('https://fe-apps.herokuapp.com/api/v1/whateverly/1901/SallyHaefling/yoga')
      .then(response => response.json())
      .then(yogaTypes => this.setState( {yogaTypes: yogaTypes.yoga.types}))
      .catch(err => console.log(err))
  }

  updateResults = query => {
    this.setState(
      {
        searchResults: search(query)
      },
      console.log(this.state.searchResults)
    );
  };

  render() {
    console.log(this.state)
    return (
      <section className="App">
        <Header />
        <Carousel yogaTypes={this.state.yogaTypes}/>
        <SearchBar updateResults={this.updateResults} />
        <Studios />
      </section>
    );
    
  }
}
