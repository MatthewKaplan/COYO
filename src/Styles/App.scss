body {
  margin: 0;
    background: #072E73;
}

a {
  color: inherit;
  text-decoration: none;
}