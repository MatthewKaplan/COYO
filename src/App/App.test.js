import React from 'react';
import App from './App';
import { shallow } from 'enzyme';

describe('App', () => {

  let mockTypes = [
    {
    name: "Hatha",
    typeId: 1,
    propsUsed: false,
    commonPoses: [
    "Vrikshasana",
    "Salabhasana",
    "Tadasana",
    "Uttanasana",
    "Adho Mukha Svanasana",
    "Setu Bandhasana"
    ],
    beginnerFriendly: true,
    purpose: "To align and calm your body, mind, and spirit in preparation for meditation."
    },
    {
    name: "Iyengar",
    typeId: 2,
    propsUsed: true,
    commonPoses: [
    "Tadasana",
    "Paripurna Navasana",
    "Adho Mukha Virasana",
    "Ustrasana",
    "Utthita Marichyasana",
    "Gomukhasana"
    ],
    beginnerFriendly: true,
    purpose: "To improve strength, awareness, mobility and stability."
    }]

  let mockStudios = [
    {
    name: "Whole Yoga",
    mainTypeOffered: 5,
    address: "1735 E 17th Ave, Denver, CO 80218",
    phoneNum: "(303)333-9642",
    openEveryday: true,
    mission: "Where body and mind become whole.",
    googleRating: 4.8,
    dropInFee: "$17.00",
    teacherTraining: true,
    link: "https://wholeyoga.com",
    img: "https://i.imgur.com/w7KfQWv.jpg"
    },
    {
    name: "Samadhi Yoga",
    mainTypeOffered: 5,
    address: "639 E 19th Ave, Denver, CO 80203",
    phoneNum: "(303)860-9642",
    openEveryday: true,
    mission: "To honor the science of yoga as a methodical process to create a lifestyle of health, happiness, and abundance.",
    googleRating: 4.7,
    dropInFee: "$20.00",
    teacherTraining: true,
    link: "https://samadhiyoga.net",
    img: "https://i.imgur.com/fkUvUxr.jpg"
    }]

  let mockMessage = 'This is a mock message';

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <App 
      studios={mockStudios}
      yogaTypes={mockTypes}
      />
    )
  });

  it('should render without errors', () => {
    const appComponent = wrapper.find("[data-test='component-app']");
    expect(appComponent.length).toBe(1);
  });

  it('should have default state', () => {
    const initialStudioState = wrapper.state('studios');
    const initialYogaState = wrapper.state('yogaTypes');
    const initialRenderedState = wrapper.state('rendered');
    const initialemptyTypeState = wrapper.state('emptyType');
    expect(initialStudioState).toEqual([]);
    expect(initialYogaState).toEqual([]);
    expect(initialRenderedState).toEqual([]);
    expect(initialemptyTypeState).toEqual('');
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should start as undefined before a user clicks on the button to find studios', () => {
    let result = wrapper.instance().storeEmptyType("");
    expect(result).toBe(undefined);
  });

  it('should store rendered studios', () => {
    expect(wrapper.state('rendered')).toEqual( [] );
    wrapper.instance().storeRendered(mockStudios);
    expect(wrapper.state('rendered')).toHaveLength(2);
  });

  it('should update yoga types', () => {
    expect(wrapper.state('yogaTypes')).toEqual( [] );
    wrapper.instance().updateTypes(mockTypes);
    expect(wrapper.state('yogaTypes')).toHaveLength(2);
  });

  it('should store message to emptyType', () => {
    expect(wrapper.state('emptyType')).toEqual( '' );
    wrapper.instance().storeEmptyType(mockMessage);
    expect(typeof wrapper.state('emptyType')).toBe('string');
  });

});
