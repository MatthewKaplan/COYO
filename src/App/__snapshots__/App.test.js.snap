// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<section
  className="App"
  data-test="component-app"
>
  <Header />
  <Carousel
    storeEmptyType={[Function]}
    storeRendered={[Function]}
    studios={Array []}
    yogaTypes={Array []}
  />
  <Search
    storeRendered={[Function]}
    studios={Array []}
  />
  <Controls
    rendered={Array []}
    storeEmptyType={[Function]}
    storeRendered={[Function]}
    studios={Array []}
  />
  <Studios
    rendered={Array []}
    studios={Array []}
  />
</section>
`;
